suite-native/discovery/src/discoveryThunks.ts

```diff
@@ -694,7 +698,28 @@ export const discoveryCheckThunk = createThunk(
 // It removes accounts for disabled networks and checks whether to start discovery and start if needed
 export const applyDiscoveryChangesThunk = createThunk(
     `${DISCOVERY_MODULE_PREFIX}/applyDiscoveryChangesThunk`,
-    (_, { dispatch }) => {
+    (_, { dispatch, getState }) => {
+        const isCoinEnablingActive = selectIsFeatureFlagEnabled(
+            getState(),
+            FeatureFlag.IsCoinEnablingActive,
+        );
+
+        // Make sure that first normal account is visible for enabled networks when coin enabling is active
+        // This might be needed in case user has View only device from before coin enabling was active
+        // in such case the first normal account can be invisible. We need to make it visible.
+        if (isCoinEnablingActive) {
+            const enabledDiscoveryNetworkSymbols = selectEnabledDiscoveryNetworkSymbols(getState());
+            enabledDiscoveryNetworkSymbols.forEach(networkSymbol => {
+                const firstNormalAccount = selectDeviceAccountsForNetworkSymbolAndAccountType(
+                    getState(),
+                    networkSymbol,
+                    NORMAL_ACCOUNT_TYPE,
+                ).filter(account => account.index === 0 && !account.visible);
```

Comment:
what about making a memoized selector from it?

Reply:
Good idea 5e9aa79893